frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -42,30 +58,201 @@ export const CSVUpload: React.FC<CSVUploadProps> = ({
               )
           );
         });
-        console.log("Validated CSV data:", results.data);
-        onDataChange(parsedData); // Pass validated data to parent
+
+        updateRubricWithCSVVersion1(parsedData);
+      },
+    });
+  };
+
+  const updateRubricWithCSVVersion1 = (data: CSVRow[]) => {
+    if (!rubric) return;
+
+    const newCriteria = data
+      .slice(1) // Skip header row
+      .map((row) => {
+        const criterionTitle = row[0]?.trim();
+        if (!criterionTitle) return null;
+
+        const criterion: Criteria = createCriterion(criterionTitle, "", 0, []);
+        for (let i = 1; i < row.length; i += 2) {
+          const points = Number(row[i]);
+          const description = row[i + 1]?.toString();
+          if (description) {
+            criterion.ratings.push(createRating(points, description));
+          }
+        }
+        criterion.updatePoints();
+        return criterion;
+      })
+      .filter(Boolean) as Criteria[];
+
+    setRubric((prevRubric) =>
+      prevRubric
+        ? {
+            ...prevRubric,
+            criteria: [...prevRubric.criteria, ...newCriteria],
+          }
+        : {
+            id: Date.now(), // Replace with a valid ID generator if needed
+            title: "Imported Rubric",
+            description: "Generated from CSV",
+            criteria: newCriteria,
+            pointsPossible: newCriteria.reduce(
+              (sum, criterion) => sum + criterion.points,
+              0,
+            ),
+            key: `rubric-${Date.now()}`, // Unique key for the rubric
+          },
+    );
+
+    closeImportCard();
+  };
+
+  // Version 2 Parsing
+  const parseCSVVersion2 = (file: File) => {
+    Papa.parse(file, {
+      header: false,
+      complete: (results) => {
+        console.log("Parsed CSV data (Version 2):", results.data);
+
+        const parsedData = results.data.filter((row): row is CSVRow => {
+          return Array.isArray(row) && typeof row[0] === "string";
+        });
+
+        updateRubricWithCSVVersion2(parsedData);
       },
     });
+  };
+
+  const updateRubricWithCSVVersion2 = (data: CSVRow[]) => {
```

Comment:
Extracting the parsing logic to a utility module would be nice and using sort of a simple strategy pattern within the component to call it.

So like in `CSVUpload.tsx` we could just call `importCsv(versionTwo, callback)` and all the parsing logic is wrapped up in another module.